File: tests/human/x/fs/two-sum.fs
let twoSum (nums:int list) (target:int) =
    let rec find i j =
        if i >= List.length nums then [-1; -1]
        elif j >= List.length nums then find (i+1) (i+2)
        elif nums.[i] + nums.[j] = target then [i; j]
        else find i (j+1)
    find 0 1

let result = twoSum [2;7;11;15] 9
printfn "%d" result.[0]
printfn "%d" result.[1]

File: tests/human/x/pas/two-sum.pas
program TwoSum;

type
  TIntArray = array of integer;

function twoSum(nums: TIntArray; target: integer): TIntArray;
var
  n,i,j: integer;
begin
  n := Length(nums);
  SetLength(Result, 2);
  for i := 0 to n-1 do
    for j := i+1 to n-1 do
      if nums[i] + nums[j] = target then
      begin
        Result[0] := i;
        Result[1] := j;
        Exit;
      end;
  Result[0] := -1;
  Result[1] := -1;
end;

var
  arr,res: TIntArray;
begin
  arr := TIntArray.Create(2,7,11,15);
  res := twoSum(arr, 9);
  Writeln(res[0]);
  Writeln(res[1]);
end.
